corehq/apps/data_dictionary/views.py

```diff
@@ -168,14 +169,90 @@ def _get_case_data(case_type):
             "properties_count": case_type.properties_count,
         }
 
-    queryset = CaseType.objects.filter(domain=domain).annotate(properties_count=Count('property'))
-    if not request.GET.get("load_deprecated_case_types", False) == "true":
-        queryset = queryset.filter(is_deprecated=False)
-    case_types_data = [_get_case_data(case_type) for case_type in queryset]
-    return JsonResponse({
-        "case_types": case_types_data,
-        "geo_case_property": geo_case_prop,
-    })
+    if case_type_name:
+        try:
+            skip = int(request.GET.get('skip', 0))
+            limit = int(request.GET.get('limit', 500))
+            if skip < 0 or limit < 0:
+                raise ValueError
+        except ValueError:
+            return JsonResponse({"error": _("skip and limit must be positive integers")}, status=400)
+
+        case_type = get_object_or_404(
+            CaseType.objects.annotate(properties_count=Count('property')),
+            domain=domain,
+            name=case_type_name
+        )
+        case_type_data = _get_case_data(case_type)
+
+        properties_queryset = CaseProperty.objects.select_related('group').filter(case_type=case_type)
+        properties_queryset = properties_queryset.order_by('group_id', 'index', 'pk')[skip:skip + limit]
+        properties_queryset = properties_queryset.prefetch_related(
+            Prefetch('allowed_values', queryset=CasePropertyAllowedValue.objects.order_by('allowed_value'))
+        )
+
+        current_url = request.build_absolute_uri()
+        links = {"self": update_url_query_params(current_url, {"skip": skip, "limit": limit})}
+        if skip:
+            links["previous"] = update_url_query_params(
+                current_url,
+                {"skip": max(skip - limit, 0), "limit": limit}
+            )
+        if case_type_data["properties_count"] > (skip + limit):
+            links["next"] = update_url_query_params(current_url, {"skip": skip + limit, "limit": limit})
+        case_type_data["_links"] = links
```

Comment:
This also feels like it could be split out into a separate method, maybe something like `update_url_params` or so.

Reply:
Addressed in 1339a9d204d8f63436b20fb79527a86e2b18c221